OK, this failure is legitimate
```
            # Some things can't be uploaded, if attempting to upload these datasets we mention why
            assert 'invalid' in dataset['misc_info'] or 'unsupported' in dataset['misc_info']
            return
        elif dataset['state'] == 'error' and 'empty' in dataset['misc_info']:
            return
        else:
            # state should be OK
            assert dataset['state'] == 'ok'
        # Check that correct datatype has been detected
        file_ext = dataset['file_ext']
        if 'false' in test_data.path:
            # datasets with false in their name are not of a specific datatype
            assert file_ext != PARENT_SNIFFER_MAP.get(expected_file_ext, expected_file_ext)
        else:
>           assert file_ext == PARENT_SNIFFER_MAP.get(expected_file_ext, expected_file_ext)
E           AssertionError: assert 'zip' == 'xlsx'
E             - zip
E             + xlsx
```

xlsx are zip compressed, so we need to move the zip sniffer lower / the xlsx sniffer higher.

--- a/lib/galaxy/config/sample/datatypes_conf.xml.sample
+++ b/lib/galaxy/config/sample/datatypes_conf.xml.sample
@@ -843,6 +843,8 @@
     <sniffer type="galaxy.datatypes.binary:Cpt"/>
     <sniffer type="galaxy.datatypes.binary:Edr"/>
     <sniffer type="galaxy.datatypes.binary:Vel"/>
+    <sniffer type="galaxy.datatypes.binary:Xlsx"/>
+    <sniffer type="galaxy.datatypes.binary:CompressedZipArchive"/>
     <sniffer type="galaxy.datatypes.annotation:Augustus"/>
     <sniffer type="galaxy.datatypes.triples:Rdf"/>
     <sniffer type="galaxy.datatypes.blast:BlastXml"/>

--- a/lib/galaxy/datatypes/binary.py
+++ b/lib/galaxy/datatypes/binary.py
@@ -249,6 +249,16 @@ class CompressedZipArchive(CompressedArchive):
         except Exception:
             return "Compressed zip file (%s)" % (nice_size(dataset.get_size()))
 
+    def sniff(self, filename):
+        with zipfile.ZipFile(filename) as zf:
+            zf_files = zf.infolist()
+            count = 0
+            for f in zf_files:
+                if f.file_size > 0 and not f.filename.startswith('__MACOSX/') and not f.filename.endswith('.DS_Store'):
+                    count += 1
+                if count > 1:
+                    return True
+
 
 class GenericAsn1Binary(Binary):
     """Class for generic ASN.1 binary format"""

